pkg/server/plugin/keymanager/gcpkms/gcpkms_test.go

```diff
@@ -0,0 +1,1478 @@
+			generateKeyReq: &keymanagerv1.GenerateKeyRequest{
+				KeyId:   spireKeyID1,
+				KeyType: keymanagerv1.KeyType_EC_P256,
+			},
+			signDataReq: &keymanagerv1.SignDataRequest{
+				KeyId: spireKeyID1,
+				Data:  sum256[:],
+				SignerOpts: &keymanagerv1.SignDataRequest_HashAlgorithm{
+					HashAlgorithm: keymanagerv1.HashAlgorithm_UNSPECIFIED_HASH_ALGORITHM,
+				},
+			},
+		},
+		{
+			name:       "usupported hash algorithm",
+			expectCode: codes.InvalidArgument,
+			expectMsg:  "hash algorithm not supported",
+			generateKeyReq: &keymanagerv1.GenerateKeyRequest{
+				KeyId:   spireKeyID1,
+				KeyType: keymanagerv1.KeyType_EC_P256,
+			},
+			signDataReq: &keymanagerv1.SignDataRequest{
+				KeyId: spireKeyID1,
+				Data:  sum256[:],
+				SignerOpts: &keymanagerv1.SignDataRequest_HashAlgorithm{
+					HashAlgorithm: 100,
+				},
+			},
+		},
+		{
+			name:       "non existing key",
+			expectCode: codes.NotFound,
+			expectMsg:  "key \"does_not_exists\" not found",
+			signDataReq: &keymanagerv1.SignDataRequest{
+				KeyId: "does_not_exists",
+				Data:  sum256[:],
+				SignerOpts: &keymanagerv1.SignDataRequest_HashAlgorithm{
+					HashAlgorithm: keymanagerv1.HashAlgorithm_SHA256,
+				},
+			},
+		},
+		{
+			name:       "pss not supported",
+			expectCode: codes.InvalidArgument,
+			expectMsg:  "the only RSA signature scheme supported is RSASSA-PKCS1-v1_5",
+			generateKeyReq: &keymanagerv1.GenerateKeyRequest{
+				KeyId:   spireKeyID1,
+				KeyType: keymanagerv1.KeyType_RSA_2048,
+			},
+			signDataReq: &keymanagerv1.SignDataRequest{
+				KeyId: spireKeyID1,
+				Data:  sum256[:],
+				SignerOpts: &keymanagerv1.SignDataRequest_PssOptions{
+					PssOptions: &keymanagerv1.SignDataRequest_PSSOptions{
+						HashAlgorithm: keymanagerv1.HashAlgorithm_SHA256,
+						SaltLength:    256,
+					},
+				},
+			},
+		},
+		{
+			name:              "sign error",
+			asymmetricSignErr: errors.New("error signing"),
+			expectCode:        codes.Internal,
+			expectMsg:         "failed to sign: error signing",
+			generateKeyReq: &keymanagerv1.GenerateKeyRequest{
+				KeyId:   spireKeyID1,
+				KeyType: keymanagerv1.KeyType_EC_P256,
+			},
+			signDataReq: &keymanagerv1.SignDataRequest{
+				KeyId: spireKeyID1,
+				Data:  sum256[:],
+				SignerOpts: &keymanagerv1.SignDataRequest_HashAlgorithm{
+					HashAlgorithm: keymanagerv1.HashAlgorithm_SHA256,
+				},
+			},
+		},
+		{
+			name:       "integrity verification error",
+			expectCode: codes.Internal,
+			expectMsg:  "error signing: response corrupted in-transit",
+			generateKeyReq: &keymanagerv1.GenerateKeyRequest{
+				KeyId:   spireKeyID1,
+				KeyType: keymanagerv1.KeyType_EC_P256,
+			},
+			signDataReq: &keymanagerv1.SignDataRequest{
+				KeyId: spireKeyID1,
+				Data:  sum256[:],
+				SignerOpts: &keymanagerv1.SignDataRequest_HashAlgorithm{
+					HashAlgorithm: keymanagerv1.HashAlgorithm_SHA256,
+				},
+			},
+			signatureCrc32C: &wrapperspb.Int64Value{Value: 1},
+		},
```

Comment:
can you add a test case with unsupported sign type?

Reply:
Sounds good. I've added a test case to test an unsupported key type in TestGenerateKey.